apps/api/v2/src/ee/bookings/controllers/bookings.controller.e2e-spec.ts

```diff
@@ -0,0 +1,205 @@
+  describe("User Authentication", () => {
+    let app: INestApplication;
+
+    let userRepositoryFixture: UserRepositoryFixture;
+
+    const userEmail = "bookings-controller-e2e@api.com";
+    let user: User;
+
+    let createdBooking: Awaited<ReturnType<typeof handleNewBooking>>;
+
+    beforeAll(async () => {
+      const moduleRef = await withAccessTokenAuth(
+        userEmail,
+        Test.createTestingModule({
+          imports: [AppModule, PrismaModule, AvailabilitiesModule, UsersModule],
+          providers: [SchedulesRepository, SchedulesService],
+        })
+      ).compile();
+
+      userRepositoryFixture = new UserRepositoryFixture(moduleRef);
+
+      user = await userRepositoryFixture.create({
+        email: userEmail,
+      });
+
+      app = moduleRef.createNestApplication();
+      bootstrap(app as NestExpressApplication);
+
+      await app.init();
+    });
+
+    it("should be defined", () => {
+      expect(userRepositoryFixture).toBeDefined();
+      expect(user).toBeDefined();
+    });
+
+    it("should create a booking", async () => {
+      const bookingStart = "2023-05-25T09:30:00.000Z";
+      const bookingEnd = "2023-05-25T10:30:00.000Z";
+      const bookingEventTypeId = 7;
+      const bookingTimeZone = "Europe/Londom";
+      const bookingLanguage = "en";
+      const bookingHashedLink = "";
+      const bookingMetadata = {};
+
+      const body = {
+        start: bookingStart,
+        end: bookingEnd,
+        eventTypeId: bookingEventTypeId,
+        timeZone: bookingTimeZone,
+        language: bookingLanguage,
+        metadata: bookingMetadata,
+        hashedLink: bookingHashedLink,
+      };
+
+      return request(app.getHttpServer())
+        .post("/api/v2/ee/bookings")
+        .send(body)
+        .expect(201)
+        .then(async (response) => {
+          const responseBody: ApiSuccessResponse<Awaited<ReturnType<typeof handleNewBooking>>> =
+            response.body;
+          expect(responseBody.status).toEqual(SUCCESS_STATUS);
+          expect(responseBody.data).toBeDefined();
+          expect(responseBody.data.user.email).toBeDefined();
+          expect(responseBody.data.id).toBeDefined();
+          expect(responseBody.data.uid).toBeDefined();
+          expect(responseBody.data.startTime).toEqual(bookingStart);
+          expect(responseBody.data.eventTypeId).toEqual(bookingEventTypeId);
+          expect(responseBody.data.user.timeZone).toEqual(bookingTimeZone);
+          expect(responseBody.data.metadata).toEqual(bookingMetadata);
+
+          createdBooking = responseBody.data;
+        });
+    });
+
+    it("should get bookings", async () => {
+      return request(app.getHttpServer())
+        .get("/api/v2/ee/bookings")
+        .then((response) => {
+          const responseBody: ApiSuccessResponse<Awaited<ReturnType<typeof getAllUserBookings>>> =
+            response.body;
+          const fetchedBooking = responseBody.data.bookings[0];
```

Comment:
I think a good one to add would be that responseBody.data.bookigns length should be just 1 because we made only 1 booking above.